.github/workflows/python.yml

```diff
@@ -40,33 +42,73 @@ jobs:
     timeout-minutes: 30
     strategy:
       matrix:
-        python-version: [3.6, 3.7]
-        tf-version: [1.15.0, 2.6.0]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        tf-version: ['tf15', 'tf26', 'tf27', 'tf28', 'tf29']
+        exclude:
+          - python-version: '3.8'
+            tf-version: 'tf15'
+          - python-version: '3.9'
+            tf-version: 'tf15'
+          - python-version: '3.10'
+            tf-version: 'tf15'
+          - python-version: '3.10'
+            tf-version: 'tf26'
+          - python-version: '3.10'
+            tf-version: 'tf27'
       fail-fast: false
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup python environment
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install requirements
         run: |
-          pip install --upgrade pip
+          pip install --upgrade pip setuptools wheel
           pip install -r ./submarine-sdk/pysubmarine/github-actions/test-requirements.txt
-      - name: Install pysubmarine with tf1 and pytorch
-        if: matrix.tf-version == '1.15.0'
+      - name: Install pysubmarine with tf1.5 and pytorch
+        if: ${{ matrix.tf-version == 'tf15' }}
         run: pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf,pytorch]
-      - name: Install pysubmarine with tf2 and pytorch
-        if: matrix.tf-version == '2.6.0'
-        run: pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf2,pytorch]
+      - name: Install pysubmarine with tf2.6 and pytorch
+        if: ${{ matrix.tf-version == 'tf26' }}
+        run: pip install --no-cache-dir tensorflow~=2.6 && pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf2,pytorch]
+      - name: Install pysubmarine with tf2.7 and pytorch
+        if: ${{ matrix.tf-version == 'tf27' }}
+        run: pip install --no-cache-dir tensorflow~=2.7 && pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf2,pytorch]
+      - name: Install pysubmarine with tf2.8 and pytorch
+        if: ${{ matrix.tf-version == 'tf28' }}
+        run: pip install --no-cache-dir tensorflow~=2.8 && pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf2,pytorch]
+      - name: Install pysubmarine with tf2.9 and pytorch
+        if: ${{ matrix.tf-version == 'tf29' }}
```

Comment:
we can change `tf26` to `2.6`, and update the task to something like 
```
- name: Install pysubmarine with tf2.6 and pytorch
        run: pip install --no-cache-dir tensorflow~=${{ matrix.tf-version }} && pip install --no-cache-dir -e ./submarine-sdk/pysubmarine/.[tf2,pytorch]
```
same for the tf28, tf29